native/src/utils/sendTrackingSignal.ts

```diff
@@ -18,7 +18,7 @@ export const sendRequest = async (signal: SignalType): Promise<void> => {
       await createTrackingEndpoint().request(signal)
     }
   } catch (e) {
-    if (fromError(e) === ErrorCode.NetworkConnectionFailed) {
+    if (e instanceof Error && fromError(e) === ErrorCode.NetworkConnectionFailed) {
```

Comment:
I think with your change to `fromError` this additional check should not be needed, right?